streams/readable-streams/async-iterator.any.js

```diff
@@ -292,20 +292,256 @@ promise_test(async t => {
+      c.error(error1);
+    }
+  });
+
   const it = s[Symbol.asyncIterator]();
-  it.next();
 
-  await promise_rejects_js(t, TypeError, it.return(), 'return() should reject');
-  assert_array_equals(s.events, ['pull']);
-}, 'calling return() while there are pending reads rejects');
+  await flushAsyncEvents();
+  await promise_rejects_exactly(t, error1, it.return('return value'));
+}, 'return() rejects if the stream has errored');
+
+promise_test(async t => {
+  let timesPulled = 0;
+  const s = new ReadableStream({
+    pull(c) {
+      if (timesPulled === 0) {
+        c.enqueue(0);
+        ++timesPulled;
+      } else {
+        c.error(error1);
+      }
+    }
+  });
+
+  const it = s[Symbol.asyncIterator]();
+
+  const iterResult1 = await it.next();
+  assert_equals(iterResult1.value, 0, '1st next() value');
+  assert_equals(iterResult1.done, false, '1st next() done');
+
+  await promise_rejects_exactly(t, error1, it.next(), '2nd next()');
+
+  const iterResult3 = await it.next();
+  assert_equals(iterResult3.value, undefined, '3rd next() value');
+  assert_equals(iterResult3.done, true, '3rd next() done');
+}, 'next() after next() reports an error');
+
+promise_test(async () => {
+  let timesPulled = 0;
+  const s = new ReadableStream({
+    pull(c) {
+      if (timesPulled === 0) {
+        c.enqueue(0);
+        ++timesPulled;
+      } else {
+        c.error(error1);
+      }
+    }
+  });
+
+  const it = s[Symbol.asyncIterator]();
+
+  const iterResults = await Promise.allSettled([it.next(), it.next(), it.next()]);
+
+  assert_equals(iterResults[0].status, 'fulfilled', '1st next() promise status');
+  assert_equals(iterResults[0].value.value, 0, '1st next() value');
+  assert_equals(iterResults[0].value.done, false, '1st next() done');
+
+  assert_equals(iterResults[1].status, 'rejected', '2nd next() promise status');
+  assert_equals(iterResults[1].reason, error1, '2nd next() rejection reason');
+
+  assert_equals(iterResults[2].status, 'fulfilled', '3rd next() promise status');
+  assert_equals(iterResults[2].value.value, undefined, '3rd next() value');
+  assert_equals(iterResults[2].value.done, true, '3rd next() done');
+}, 'next() after next() reports an error, no awaiting');
+
+promise_test(async t => {
+  let timesPulled = 0;
+  const s = new ReadableStream({
+    pull(c) {
+      if (timesPulled === 0) {
+        c.enqueue(0);
+        ++timesPulled;
+      } else {
+        c.error(error1);
+      }
+    }
+  });
+
+  const it = s[Symbol.asyncIterator]();
+
+  const iterResult1 = await it.next();
+  assert_equals(iterResult1.value, 0, '1st next() value');
+  assert_equals(iterResult1.done, false, '1st next() done');
+
+  await promise_rejects_exactly(t, error1, it.next(), '2nd next()');
+
+  const iterResult3 = await it.return('return value');
+  assert_equals(iterResult3.value, 'return value', 'return() value');
+  assert_equals(iterResult3.done, true, 'return() done');
+}, 'return() after next() reports an error');
```

Comment:
nit: the English phrase has a bit of an ambiguity. I first read it as "[return() after next()] reports an error", only afterwards did I realize it's actually "return() after [next() reports an error]". This also applies to the "next() after next()" tests above.